News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/adapter/ViewHolder.java

```diff
@@ -304,6 +335,37 @@ public void setRssItem(RssItem rssItem) {
         }
     }
 
+    private static void scaleTextSize(TextView tv, int initialSize, boolean halfScale) {
+        SharedPreferences mPrefs = PreferenceManager.getDefaultSharedPreferences(tv.getContext());
+        float scalingFactor = Float.parseFloat(mPrefs.getString(SettingsActivity.SP_FONT_SIZE, "1.0"));
+        if(initialSize < 0) {
+            initialSize = Math.round(tv.getTextSize());
+        }
+        if(halfScale) {
+            scalingFactor = scalingFactor + (1-scalingFactor)/2;
+        }
+        // float sp = initialSize / tv.getContext().getResources().getDisplayMetrics().scaledDensity;  // transform scaled pixels, device pixels
+        tv.setTextSize(TypedValue.COMPLEX_UNIT_PX, Math.round(initialSize*scalingFactor));
+    }
+
+    private static int scaleSimpleTextLines(TextView tv) {
+        SharedPreferences mPrefs = PreferenceManager.getDefaultSharedPreferences(tv.getContext());
+        float scalingFactor = Float.parseFloat(mPrefs.getString(SettingsActivity.SP_FONT_SIZE, "1.0"));
+        int numLines = 5;
+
+        if(scalingFactor < 0.9) {
+            numLines = 7;
```

Comment:
As some of the text (last line) was cut off for me when using the "small" font, I suggest we use 6 lines here? Or did it work for you properly? :)

```suggestion
            numLines = 6;
```